sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusConnection.cs

```diff
@@ -0,0 +1,533 @@
+        {
+            switch (options.TransportType)
+            {
+                case ServiceBusTransportType.AmqpTcp:
+                case ServiceBusTransportType.AmqpWebSockets:
+                    return new AmqpClient(fullyQualifiedNamespace, entityName, credential, options);
+
+                default:
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources1.InvalidTransportType, options.TransportType.ToString()), nameof(options.TransportType));
+            }
+        }
+
+        /// <summary>
+        ///   Builds the audience for use in the signature.
+        /// </summary>
+        ///
+        /// <param name="transportType">The type of protocol and transport that will be used for communicating with the Service Bus service.</param>
+        /// <param name="fullyQualifiedNamespace">The fully qualified Service Bus namespace.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="entityName">The name of the specific entity to connect the client to.</param>
+        ///
+        /// <returns>The value to use as the audience of the signature.</returns>
+        ///
+        private static string BuildAudienceResource(ServiceBusTransportType transportType,
+                                                    string fullyQualifiedNamespace,
+                                                    string entityName)
+        {
+            var builder = new UriBuilder(fullyQualifiedNamespace)
+            {
+                Scheme = transportType.GetUriScheme(),
+                Path = entityName,
+                Port = -1,
+                Fragment = string.Empty,
+                Password = string.Empty,
+                UserName = string.Empty,
+            };
+
+            if (builder.Path.EndsWith("/"))
+            {
+                builder.Path = builder.Path.TrimEnd('/');
+            }
+
+            return builder.Uri.AbsoluteUri.ToLowerInvariant();
+        }
+
+        ///// <summary>
+        /////   Performs the actions needed to validate the set of properties for connecting to the
+        /////   Service Bus service, as passed to this client during creation.
+        ///// </summary>
+        /////
+        ///// <param name="properties">The set of properties parsed from the connection string associated this client.</param>
+        ///// <param name="connectionStringArgumentName">The name of the argument associated with the connection string; to be used when raising <see cref="ArgumentException" /> variants.</param>
+        /////
+        ///// <remarks>
+        /////   In the case that the properties violate an invariant or otherwise represent a combination that
+        /////   is not permissible, an appropriate exception will be thrown.
+        ///// </remarks>
+        /////
+        //private static void ValidateConnectionProperties(ConnectionStringProperties properties,
```

Comment:
you absolutely want this;   In the case where you have a connection string with the Service Bus entity and you also have one provided by the constructor, this was intended to be sure they were the same and complain if they differ.

This is a necessary convenience because of the dumpster fire user experience for getting connection strings from the portal.

Reply:
Covered in https://github.com/Azure/azure-sdk-for-net/issues/9884